mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -471,6 +484,59 @@ class LinearRegressionSummary private[regression] (
     predictions.select(t(col(predictionCol), col(labelCol)).as("residuals"))
   }
 
+  lazy val numInstances: Long = predictions.count()
+
+  lazy val dfe = if (model.getFitIntercept) {
+    numInstances - model.weights.size -1
+  } else {
+    numInstances - model.weights.size
+  }
+
+  lazy val devianceResiduals: Array[Double] = {
+    val weighted = if (model.getWeightCol.isEmpty) lit(1.0) else sqrt(col(model.getWeightCol))
+    val dr = predictions.select(col(model.getLabelCol).minus(col(model.getPredictionCol))
+      .multiply(weighted).as("weightedResiduals"))
+      .select(min(col("weightedResiduals")).as("min"), max(col("weightedResiduals")).as("max"))
+      .take(1)(0)
+    Array(dr.getDouble(0), dr.getDouble(1))
+  }
+
+  lazy val seCoef: Array[Double] = {
```

Comment:
`coefficientStandardErrors`? It is hard to guess what `seCoef` means. In the doc, we should say "intercept" is not supported.